build-tools/xamarin-android-docimporter-ng/Xamarin.Android.ApiTools.ParameterNameExtractor/Xamarin.Android.ApiTools.JavaStubImporter/JavaStubParser.cs

```diff
@@ -243,9 +245,9 @@ public JavaStubGrammar ()
 			type_members.Rule = MakeStarRule (type_members, type_member);
 			type_member.Rule = nested_type_decl | ctor_decl | method_decl | field_decl | static_ctor_decl;
 			nested_type_decl.Rule = type_decl;
-			opt_enum_member_initializers.Rule = Empty | enum_member_initializers + ";";
+			enum_members_decl.Rule = enum_member_initializers + ";";
 			enum_member_initializers.Rule = MakeStarRule (enum_member_initializers, ToTerm (","), enum_member_initializer);
-			enum_member_initializer.Rule = identifier + "(" + ")";
+			enum_member_initializer.Rule = annotations + identifier + "(" + ")";
```

Comment:
Similarly, this looks to me like the `()` is required, but they're optional.

```java
enum Day {
  SUNDAY
}
```

is perfectly valid, and contains no parenthesis.

Related? https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

Reply:
Note that it is NOT A FULL JAVA PARSER.

Reply:
Yes, not a full Java parser.

Doesn't it need to parse *enough* that it doesn't get confused when it finally hits methods? Doesn't that "enough" need to be sufficiently complete to allow getting that far?